torchtext/utils.py

```diff
@@ -75,3 +88,31 @@ def unicode_csv_reader(unicode_csv_data, **kwargs):
 def utf_8_encoder(unicode_csv_data):
     for line in unicode_csv_data:
         yield line.encode('utf-8')
+
+
+def extract_archive(from_path, to_path=None, remove_finished=False):
```

Comment:
Add a flag like "format=tar" to indicate it's using tar so we can extend it later on. This is also very useful in general.

Reply:
We can simply extend it by adding the other branches between "if" and "else"